cfgmgr/timerangemgr.cpp

```diff
@@ -0,0 +1,189 @@
+    if (runOnce)
+    {
+        // Delete the time range configuration entry after the task has been disabled
+        // writeCrontabFile() will delete the crontab file itself after the task has been executed
+        command_disabled += " ; /usr/bin/redis-cli -n " + to_string(CONFIG_DB) + " del '" + CFG_TIME_RANGE_TABLE_NAME + "|" + taskName + "'";
+    }
+    // {
+    //     stringstream ss;
+    //     ss << "/usr/bin/redis-cli -n " << STATE_DB << " HSET '" << STATE_TIME_RANGE_STATUS_TABLE_NAME << "|" << taskName << "' '" << TIME_RANGE_STATUS_STR << "' '" << TIME_RANGE_DISABLED_STR << "'";
+    //     if (runOnce){
+    //         // Delete the time range configuration entry after the task has been disabled
+    //         // writeCrontabFile() will delete the crontab file itself after the task has been executed
+    //         ss << " ; /usr/bin/redis-cli -n " << CONFIG_DB << " del '" << CFG_TIME_RANGE_TABLE_NAME << "|" << taskName << "'";
+    //     }
+    //     command_disabled = ss.str();
+    // }
+
+    // Service file for enabling the task
+    if (writeCrontabFile(enableCrontabName, startTime, command_enabled, runOnce) != task_process_status::task_success)
+    {
+        return task_process_status::task_need_retry;
+    }
+
+    // Service file for disabling the task
+    if (writeCrontabFile(disableCrontabName, endTime, command_disabled, runOnce) != task_process_status::task_success)
+    {
+        return task_process_status::task_need_retry;
+    }
+
+    SWSS_LOG_INFO("Succesfully created crontab files for %s", taskName.c_str());
+
+    return task_process_status::task_success;
+}
+
+task_process_status TimeRangeMgr::doTimeRangeTask(const string &rangeName, const vector<FieldValueTuple> &fieldValues)
+{
+    SWSS_LOG_ENTER();
+    string start = "";
+    string end = "";
+    string runOnce = "";
+
+    // Set time range status to disabled by default
+    vector<FieldValueTuple> stateTableFieldValues;
+    string key = rangeName;
+    stateTableFieldValues.emplace_back(FieldValueTuple(TIME_RANGE_STATUS_STR, TIME_RANGE_DISABLED_STR));
+
+    for (const auto &i : fieldValues)
+    {
+        if (fvField(i) == "start")
+        {
+            start = fvValue(i);
+        }
+        else if (fvField(i) == "end")
+        {
+            end = fvValue(i);
+        }
+        else if (fvField(i) == "runOnce")
+        {
+            runOnce = fvValue(i);
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Time range %s has unknown field %s", rangeName.c_str(), fvField(i).c_str());
+            // Can skip instead of returning invalid entry
+            return task_process_status::task_invalid_entry;
+        }
+    }
+
+    if (start == "" || end == "")
+    {
+        SWSS_LOG_ERROR("Time range %s is missing start or end time", rangeName.c_str());
+        return task_process_status::task_invalid_entry;
+    }
+
+    // Create systemd files for time range and enable them
+    // TODO sanitize inputs
```

Comment:
yes, verify time ranges validity.

Reply:
Need to strictly validate and sanitize time range since this can be a security vulnerability since we are running cron commands as root (albeit only in swss container).